File: phten/PTApp/PTApp/PTPages/Verify/Identify/Service/PTUploadIDImageService.m
//
//  PTUploadIDImageService.m
//  PTApp
//
//  Created by Jacky on 2024/8/21.
//

#import "PTUploadIDImageService.h"

#import "AFNetworking.h"
@interface PTUploadIDImageService()
@property (nonatomic, strong) UIImage *image;
@property (nonatomic, strong) NSDictionary *dic;

@end
@implementation PTUploadIDImageService

- (id)initWithImage:(UIImage *)image param:(NSDictionary *)dic {
    self = [super init];
    if (self) {
        _image = image;
        _dic = dic;
    }
    return self;
}

- (YTKRequestMethod)requestMethod {
    return YTKRequestMethodPOST;
}

- (NSString *)requestUrl {
    return PTUploadOcr;
}
- (BOOL)isShowLoading
{
    return YES;
}
- (AFConstructingBlock)constructingBodyBlock {
    
    WEAKSELF
    return ^(id<AFMultipartFormData> formData) {
        NSData *data = [UIImage scaleBiteDataImage:weakSelf.image toKBite:1024];
        NSString *formKey = @"am";
        NSString *name = @"imageFile.jpg";
        NSString *type = @"image/jpeg";

        [formData appendPartWithFileData:data name:formKey fileName:name mimeType:type];
    };
}

- (NSString *)responseFileId {
    NSDictionary *dict = self.responseJSONObject[@"resultObject"];
    return dict[@"file_id"];
}
- (id)requestArgument
{
    return _dic ? : @{};
}
@end

File: phten/PTApp/PTApp/PTPages/Verify/Identify/Service/PTUploadIDImageService.swift
//
//  PTUploadIDImageService.swift
//  PTApp
//

import UIKit

@objc(PTUploadIDImageService)
class PTUploadIDImageService: PTImageUploadRequest {
    @objc(initWithImage:param:)
    init(image: UIImage, param params: NSDictionary?) {
        super.init(
            image: image,
            params: params,
            path: PTAPIEndpoint.uploadOCR,
            fileFieldName: "am",
            fileName: "imageFile.jpg",
            showLoading: true
        )
    }

    @objc var responseFileId: String? {
        guard let json = responseJSONObject as? [String: Any],
              let result = json["resultObject"] as? [String: Any] else {
            return nil
        }
        return result["file_id"] as? String
    }
}
